### PR TITLE
Iss419 brighthub get data update url

### PR DESCRIPTION
**Pull request by stephenholleran**
Please refer to issue #419.

### CODE REVIEW

File: brightwind/load/load.py [hunk 1]
```diff
@@ -1357,70 +1361,35 @@ def get_data(measurement_station_uuid, date_from=None, date_to=None):
...
+                raise ValueError("{}. {}".format(response_json['error'], response_json['details']))
+
+        if response.status_code == 404 and 'details' in response_json:  # other BrightHub specific errors.
+            raise ValueError("{}. {}".format(response_json.get('error', ''), response_json['details']))
+        elif 'error' in response_json:
+            raise ValueError("Unexpected error: Status Code {}. {}".format(response.status_code, response.text))
+
+        pre_signed_url = response_json["url"]
+        timeseries_response = requests.get(pre_signed_url)
+        df = pd.read_csv(StringIO(timeseries_response.text, ))
```
Comment (dancasey-ie) [inline brightwind/load/load.py:1391]: remove empty second property

File: brightwind/load/load.py [hunk 2]
```diff
@@ -1357,70 +1361,35 @@ def get_data(measurement_station_uuid, date_from=None, date_to=None):
...
+            # raise the 503 error if all the attempts have been used up
+            if response.status_code == 503 and assembly_in_progress_err_msg in response_json.get('details'):
+                raise ValueError("{}. {}".format(response_json['error'], response_json['details']))
+
+        if response.status_code == 404 and 'details' in response_json:  # other BrightHub specific errors.
+            raise ValueError("{}. {}".format(response_json.get('error', ''), response_json['details']))
+        elif 'error' in response_json:
+            raise ValueError("Unexpected error: Status Code {}. {}".format(response.status_code, response.text))
+
+        pre_signed_url = response_json["url"]
```
Comment (dancasey-ie) [inline brightwind/load/load.py:1389]: call presigned_url instead of pre_signed_url to match BH code

File: brightwind/load/load.py [hunk 3]
```diff
@@ -1357,70 +1361,35 @@ def get_data(measurement_station_uuid, date_from=None, date_to=None):
...
+        timeseries_response = requests.get(pre_signed_url)
```
Comment (dancasey-ie) [inline brightwind/load/load.py:1390]: Consider error handling presigned_url request.

But the only one that comes to mind is if the presigned_url has expired but that should not happen when running this code. Or of there is an internet error

**Review by dancasey-ie — CHANGES_REQUESTED** [review]

File: brightwind/load/load.py (hunk 1 above)
Comment (stephenholleran) [reply brightwind/load/load.py:1391]: I think you put that in in the old chunking code. See line 1416 of what was deleted. I'll try it without it.

**Comment by stephenholleran** [comment]
Thanks @dancasey-ie.

**Comment by rach185** [comment]
I'm now testing this from bhdev. The date format that is expected isn't what i would expect, do you mean for the user to input yyyy-mm-ddThh:mm:ss? 

image

See the change to `date_from` which allows me to pull data.
image

**Comment by stephenholleran** [comment]
> do you mean for the user to input yyyy-mm-ddThh:mm:ss?
No I don't.
It should work the same as all other brightwind functions where the date could be "YYYY-MM-DD" or "YYYY-MM-DD hh:mm:ss". We might need to convert the date a user puts in to the one with the "T" or else we should ask BrightHub to change. @dancasey-ie thoughts?

Also, you can call it just using ```bw.LoadBrightHub.get_data()```

**Comment by dancasey-ie** [comment]
the brighthub GET /measurements-locations/{uuid}/timeseries-data API is set up to accept dates in the format  "YYYY-MM-DD hh:mm:ss" (no T) or "YYYY-MM-DDThh:mm:ss" (with T). 

Maybe brighthub should allow for dates with no time component, but it is not Brighthub convention so I do not think we will have in this release. @stephenholleran unless you think its a priority.

Alternatively, convert date strings to datetime strings using the datetime library before calling the brighthub api inside the brightwind library

**Comment by stephenholleran** [comment]
Thanks @dancasey-ie.

@rach185 I think the best approach is to convert the date into a datetime within the brightwind library using the datetime library. So "_2016-04-01_" would become "_2016-04-01 00:00:00_". This could be a sub function so can be used else where. Are you able to implement this?

Thanks,

**Comment by rach185** [comment]
I've created a subfunction, `__reformat_datetime_str`, in the LoadBrightHub class and updated the `__get_timeseries_data` function to use this.

**Comment by stephenholleran** [comment]
Thanks @rach185.
I just did a bit of refactoring on it. Moved it up above the calling function and renamed to to say what it does.

Are you happy with everything else? If you are we'll merge.

**Comment by rach185** [comment]
Thanks @stephenholleran. Yep I'm happy with everything else!

**Pull request merged**